test/e2e.test.ts

```diff
@@ -37,4 +37,33 @@ describe('turnstile', async () => {
       )
     )
   })
+
+  it('expects no callback loop', async () => {
```

Comment:
This test also runs fine without the fix applied when it should error instead :speak_no_evil:

Reply:
oh... will try to make it fail 👀

Reply:
had to increase to 2500, even 2000 isn't enough

Reply:
Hmm, that's test flakiness I'd like to avoid. What do you think about this test?

```ts
it('expects no callback loop', async () => {
  const page = await createPage()
  const logs: string[] = []

  page.on('console', event => {
    logs.push(event.text())
  })

  await page.goto(url('/'))
  await page.click('button')
  await page.click('#home-link')
  await page.waitForEvent('console')

  expect(logs).toHaveLength(0)
})
```

Reply:
I've pushed the simplification of this test and the one above, removing the discouraged usage of waiting for `networkidle`. That push can be reverted of course if necessary, just made to keep things speedy^^